### PR TITLE
BEP-6: Add delist proposal

### PR DESCRIPTION
**Pull request by HaoyangLiu**

### CODE REVIEW

File: BEP6.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+To accelerate dex TPS and make the community healthier, some tokens with bad reputations or some trading pairs have very little trading volume should be removed.
+
+## 5. Specification
+
+To delist a trading pair, users just need to to send a `DelistTradingPair` proposal. In the proposal description field, users should specify which trading pairs will be delisted and why to do this. If the proposal is passed before the time limit, then target trading pairs will be delisted and all related orders will be cancelled.
+
+### 5.1 Send proposal
+
+Proposal should contain:
```
Comment (huangsuyu) [inline BEP6.md]: i think we need to add message types here.

File: BEP6.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+1. Trading pair to delist: base asset and quote asset
+2. Justification: why we need to delist this token.
+
+### 5.2 Delist trading pairs and expire all orders in breathe block
+We assume that the ideal proposal life cycle is like this:
+- The deposit period is 2 days
+- The voting period is 14 days
+- The delisting operation will happen 3 days after proposal is passed. 
+
+The total time is 20 days. In each breathe block, all `DelistTradingPair` proposals which are submitted in less than 20 days ago will be judged. If a `DelistTradingPair` proposal is passed, and its passed time is 3 day before the breathe block time, then a delist operation will be triggered.
```
Comment (yutianwu) [inline BEP6.md]: We do not need to mention the details

Comment (yutianwu) [reply BEP6.md]: Just mention that trading pair will be delisted after three days

File: BEP6.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+|                 |       | deposit period |        | voting period |        | delayed days |       |        |
+|                 |       |   +--------+   |        | +-----------+ |        |  +--------+  |       |        |
+| Submit proposal | <---> |   | 2 days |   | <----> | | less than | | <----> |  | 3 days |  | <---> | delist |
+|                 |       |   +--------+   |        | |  14 days  | |        |  +--------+  |       |        |
+|                 |       |                |        | +-----------+ |        |              |       |        |
++-----------------+       +----------------+        +---------------+        +--------------+       +--------+
+```
+If a `DelistTradingPair` proposal is passed, and its passed time is 3 day before the breathe block time, then a delist operation will be triggered.
+
+When a trading pair is delisted, all related uncompleted orders will be expired. The locked tokens in the orders will be refunded to users and no expire fee will be deducted.
```
Comment (yutianwu) [inline BEP6.md]: fee will be charged as normal expire orders

File: BEP6.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# BEP-5: Delist Trading Pairs
+
```
Comment (yutianwu) [inline BEP6.md]: should be BEP-6

File: BEP6.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
+# BEP-6: Delist Trading Pairs
+
+## 1. Summary
+
+This BEP describes a proposal for delisting trading pairs on the Binance Chain.
+
+## 2. Abstract
+
+Currently, we already have listing proposal to list new trading pairs. However, we can't delist any trading pairs now. Suppose a token has very bad reputation or one of its trading pairs has little trading volume for a long time, then the community might consider to remove related trading paris. BIP-5 proposal just provides the community a method to delist the unwanted trading pairs.
```
Comment (notatestuser) [inline BEP6.md]: `paris` - typo

File: BEP6.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
+# BEP-6: Delist Trading Pairs
+
+## 1. Summary
+This BEP describes a proposal for delisting trading pairs on the Binance Chain.
```
Comment (darren-liu) [inline BEP6.md]: I suggest "delisting trading pairs from the Binance DEX."

File: BEP6.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+|    VotingPeriod   |   int      | Length of the voting period|
+
+For other document about governance proposal, please refer to this page: https://docs.binance.org/governance.html#proposal-parameters
+
+### 5.2 Cooling-Off Period
+Once a `DelistTradingPair` proposal is passed, it will enter cooling-off period. This period length is around 3 days from the time of vote pass until the 3rd UTC midnight. As the name suggests, users should re-evaluate the market and take action on the asset to be delisted as they wish, as in this period, users can create new orders and cancel orders on the trading pair as usual. Once this period is ended, all outstanding orders (unfilled or uncanceled) on the market will expire.
+### 5.3 Delist
+Delist happens on the 3rd midnight (UTC) after the proposal is voted pass. There are 2 steps to delist trading pairs as below.
+#### 5.3.1 Trading Pairs Constraint Check
+In binance chain, all listed tokens must be listed against BNB first. Suppose there are three tokens on binance chain: `A`, `B` and `BNB`, and there only one trading pair: `A_BNB`. If someone wants to create a new trading pair: `A_B`, then he must create another trading pair first: `B_BNB`.
```
Comment (darren-liu) [inline BEP6.md]: `On Binance Chain`

**Review by jonesk7734 — APPROVED** [review]
😀

**Pull request merged**